tests/conftest.py

```diff
@@ -1,12 +1,42 @@
-def pytest_addoption(parser):
-    """
-    Add CLI options to `pytest` to pass those options to the test cases.
-    These options are used in `pytest_generate_tests`.
+from pathlib import Path
+from threading import Thread
+from typing import Any, Iterator
+
+import pytest
+from click.testing import CliRunner
+
+from . import PatroniAPI
+
+
+@pytest.fixture(
+    params=[False, True],
+    ids=lambda v: "new-replica-state" if v else "old-replica-state",
+)
+def old_replica_state(request: Any) -> Any:
+    return request.param
+
+
+@pytest.fixture(scope="session")
+def datadir() -> Path:
+    return Path(__file__).parent / "json"
+
+
+@pytest.fixture(scope="session")
+def patroni_api(
+    tmp_path_factory: pytest.TempPathFactory, datadir: Path
+) -> Iterator[PatroniAPI]:
+    """A fake HTTP server for the Patroni API serving files from a temporary
+    directory.
     """
-    parser.addoption("--use-old-replica-state", action="store_true", default=False)
+    httpd = PatroniAPI(tmp_path_factory.mktemp("api"), datadir=datadir)
+    t = Thread(target=httpd.serve_forever)
+    t.start()
+    yield httpd
+    httpd.shutdown()
+    t.join()
 
 
-def pytest_generate_tests(metafunc):
-    metafunc.parametrize(
-        "use_old_replica_state", [metafunc.config.getoption("use_old_replica_state")]
-    )
+@pytest.fixture
+def runner() -> CliRunner:
+    """A CliRunner with stdout and stderr not mixed."""
+    return CliRunner(mix_stderr=False)
```

Comment:
Is `mix_stderr` used to make things easier when debugging ?

Reply:
In fact, it's needed as long as we use standard streams for other purposes, typically logging as done in next commits (this is explained in the message of ea92809cb333ea8f7de916eccd6549aa9921133a).